examples/dmd/heat_conduction_dmdc.cpp

```diff
@@ -0,0 +1,1028 @@
+        sout << "parallel " << num_procs << " " << myid << endl;
+        int good = sout.good(), all_good;
+        MPI_Allreduce(&good, &all_good, 1, MPI_INT, MPI_MIN, pmesh->GetComm());
+        if (!all_good)
+        {
+            sout.close();
+            visualization = false;
+            if (myid == 0)
+            {
+                cout << "Unable to connect to GLVis server at "
+                     << vishost << ':' << visport << endl;
+                cout << "GLVis visualization disabled.\n";
+            }
+        }
+        else
+        {
+            sout.precision(precision);
+            sout << "solution\n" << *pmesh << u_gf;
+            sout << flush;
+        }
+    }
+
+#ifdef MFEM_USE_GSLIB
+    if (pointwiseSnapshots)
+    {
+        pws->GetSnapshot(u_gf, pwsnap);
+
+        ostringstream dmd_filename;
+        dmd_filename << "snap_" << to_string(radius) << "_" << to_string(alpha)
+                     << "_" << to_string(cx) << "_" << to_string(cy) << "_0";
+        if (myid == 0)
+        {
+            cout << "Writing DMD snapshot at step 0, time 0.0" << endl;
+            pwsnap_CAROM->write(dmd_filename.str());
+        }
+    }
+#endif
+
+    StopWatch fom_timer, dmd_training_timer, dmd_prediction_timer;
+
+    fom_timer.Start();
+
+    // 10. Perform time-integration (looping over the time iterations, ti, with a
+    //     time-step dt).
+    ode_solver->Init(oper);
+    double t = 0.0;
+    vector<double> ts;
+    CAROM::Vector* init = NULL;
+
+    CAROM::Database *db = NULL;
+    if (csvFormat)
+        db = new CAROM::CSVDatabase();
+    else
+        db = new CAROM::HDFDatabase();
+
+    vector<int> snap_list;
+
+    fom_timer.Stop();
+
+    CAROM::DMD* dmd_u = NULL;
+
+    if (offline)
+    {
+        dmd_training_timer.Start();
+
+        // 11. Create DMD object and take initial sample.
+        u_gf.SetFromTrueDofs(u);
+        dmd_u = new CAROM::DMD(u.Size(), dt);
+        dmd_u->takeSample(u.GetData(), t);
+
+        if (myid == 0)
+        {
+            std::cout << "Taking snapshot at: " << t << std::endl;
+        }
+
+        dmd_training_timer.Stop();
+    }
+    if (online)
+    {
+        u_gf.SetFromTrueDofs(u);
+        init = new CAROM::Vector(u.GetData(), u.Size(), true);
+    }
+
+    if (save_dofs && myid == 0)
+    {
+        if (csvFormat)
+        {
+            mkdir((outputPath + "/step0").c_str(), 0777);
+            db->putDoubleArray(outputPath + "/step0/sol.csv", u.GetData(), u.Size());
+        }
+        else
+        {
+            db->create(outputPath + "/dmd_0.hdf");
+            db->putDoubleArray("step0sol", u.GetData(), u.Size());
+        }
+    }
+
+    ts.push_back(t);
+    snap_list.push_back(0);
+
+    bool last_step = false;
+    for (int ti = 1; !last_step; ti++)
+    {
+        fom_timer.Start();
+
+        if (t + dt >= t_final - dt/2)
+        {
+            last_step = true;
+        }
+
+        ode_solver->Step(u, t, dt);
```

Comment:
This is what I'm referring to in the earlier comment; I don't think this needs to be done in the online phase?

Reply:
This is necessary for computing the reference solution at the testing parameter. Arguably, it is not part of the online phase in a ROM setting. But, as in other libROM examples on MFEM, I guess we want to minimize the number of phases to get the results.